sno/cli_util.py

```diff
@@ -74,6 +76,29 @@ def convert(self, value, param, ctx):
         return value
 
 
+class JsonFromFile(StringFromFile):
+    name = 'json'
+
+    def __init__(self, schema=None, **file_kwargs):
+        super().__init__(**file_kwargs)
+        self.schema = schema
+
+    def convert(self, value, param, ctx):
+        value = super().convert(value, param, ctx)
+        try:
+            value = json.loads(value)
+        except json.JSONDecodeError as e:
+            self.fail(
+                f"Invalid JSON: {e}", param, ctx,
+            )
+        if self.schema:
+            try:
+                jsonschema.validate(instance=value, schema=self.schema)
+            except jsonschema.ValidationError as e:
+                raise click.BadParameter(str(e), ctx=ctx, param=param,) from e
```

Comment:
why does this error case use an exception and the other one use `self.fail()`?

Reply:
not sure, i'll make this use self.fail() (the exception is ultimately BadParameter anyway)